camel/storages/vectordb_storages/milvus.py

```diff
@@ -0,0 +1,360 @@
+
+        Returns:
+            str: A unique collection name.
+        """
+        return datetime.now().isoformat()
+
+    def _get_collection_info(self, collection_name: str) -> Dict[str, Any]:
+        r"""Retrieves details of an existing collection.
+
+        Args:
+            collection_name (str): Name of the collection to be checked.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing details about the
+                collection.
+        """
+        vector_count = self._client.num_entities(collection_name)
+
+        collection_info = self._client.describe_collection(collection_name)
+        collection_id = collection_info['collection_id']
+        dim_value = next(
+            (field['params']['dim']
+             for field in collection_info['fields'] if field['description'] ==
+             'The numerical representation of the vector'), None)
+
+        return {
+            # Return the id of the collection.
+            "id": collection_id,
+            # Return the number of vector.
+            "vector_count": vector_count,
+            # Return the dimension of vector.
+            "vector_dim": dim_value,
+        }
+
+    def _validate_and_convert_vectors(
+            self, records: List[VectorRecord]) -> List[dict]:
+        r"""
+        Validates and converts VectorRecord instances to the format expected
+        by Milvus.
+
+        Args:
+            records (List[VectorRecord]): List of vector records to validate
+            and convert.
+
+        Returns:
+            List[dict]: A list of dictionaries formatted for Milvus insertion.
+        """
+
+        validated_data = []
+
+        for record in records:
+            if record.payload is not None:
+                record_dict = {
+                    "id": record.id,
+                    "payload": record.payload['message'],
+                    "vector": record.vector,
+                }
+                validated_data.append(record_dict)
+            else:
+                record_dict = {
+                    "id": record.id,
+                    "payload": '',
+                    "vector": record.vector,
+                }
+                validated_data.append(record_dict)
+
+        return validated_data
+
+    def add(
+        self,
+        records: List[VectorRecord],
+        **kwargs,
+    ) -> None:
+        r"""Adds a list of vectors to the specified collection.
+
+        Args:
+            records (List[VectorRecord]): List of vectors to be added.
+            **kwargs (Any): Additional keyword arguments.
+
+        Raises:
+            RuntimeError: If there was an error in the addition process.
+        """
+        validated_records = self._validate_and_convert_vectors(records)
+
+        op_info = self._client.insert(
+            collection_name=self.collection_name,
+            data=validated_records,
+            **kwargs,
+        )
+        self._client.flush(collection_name=self.collection_name)
```

Comment:
We don't recommend to manually flush the collection. Manual flushes will cause too many small data segments to be compacted and then leads to high cost of disk space.
If you concern about search with latest data, you can set consistency_level in search method.

Reply:
thanks for the insight, have removed `flush`